bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/findandreplace/IFindReplaceLogicStatus.java

```diff
@@ -0,0 +1,10 @@
+package org.eclipse.ui.findandreplace;
+
+public interface IFindReplaceLogicStatus {
+
+	/**
+	 * @return The string that can be used to display the message to the user.
+	 */
+	public String getMessage();
```

Comment:
The status object should not contain the functionality to provide a message. The status of the functionality could be visualized by different UIs in different ways (textual representations, graphical representations etc.), which is why converting the actual status into a UI representation (such as a text message) should not be part of the status object but of some separate status-to-message conversion that defines for all status objects how they are textually represented.